### ISSUE TITLE
Can't settle

### ISSUE THREAD

**Issue by bonomat** [itchysats/itchysats#1298]
Tested on master: ffadfca0efff1571855fc5f583dda84202b0c70b  

I had the taker running over night, connection is up with the maker and I receive heartbeats. 

I can't close my positions and get the following error: 

```
Price feed not available: Actor address disconnected
```

![image](https://user-images.githubusercontent.com/224613/152055490-dc634c76-6f3b-487c-b33a-24ba0270f973.png)

Price information seems to be hear because all positions have P/L

![image](https://user-images.githubusercontent.com/224613/152055582-d7af66be-ef14-45de-8fa3-857fe9ca81a7.png)

Nothing in the logs

**Comment by thomaseizinger** [comment]
> Nothing in the logs

Do you still see the bitmex pings?

**Comment by bonomat** [comment]
> > Nothing in the logs
> 
> Do you still see the bitmex pings?

Can't say I noticed them but I just restarted the terminal and the logs are gone.

**Comment by da-kami** [comment]
Not sure it helps, but this might be the same/related to https://github.com/itchysats/itchysats/issues/1137

**Comment by thomaseizinger** [comment]
> Can't say I noticed them but I just restarted the terminal and the logs are gone.

They are only logged on `TRACE`.

I am baffled on how this actor can be disconnected. We catch all errors of the websocket task and report the failure to the supervisor which restarts us.

**Comment by da-kami** [comment]
I ran into this scenario:

```
2022-02-02 03:34:06 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:06 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:11 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:12 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:19 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:20  INFO daemon::maker_inc_connections: Upgrade new connection taker_address=127.0.0.1:59859
2022-02-02 03:34:20 DEBUG daemon::noise: Noise protocol responder handshake is complete
2022-02-02 03:34:20  INFO daemon::maker_inc_connections: Connection upgrade successful taker_id=ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442 taker_address=127.0.0.1:59859
2022-02-02 03:34:20  INFO daemon::maker_inc_connections: Connection is ready taker_id=ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442
2022-02-02 03:34:20 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:20 TRACE daemon::monitor: Updating status of 0 transactions
2022-02-02 03:34:25 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:25 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:31 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:31 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:34  INFO rocket::server: POST /api/order/sell application/json:    
2022-02-02 03:34:34 TRACE daemon::maker_inc_connections: Sent new order: Some(OrderId(Hyphenated(137c7cdb-789a-4e6e-98c2-eb0b6f4e99ea))) taker_id=ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442
2022-02-02 03:34:36 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:36 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:40 TRACE daemon::monitor: Updating status of 0 transactions
2022-02-02 03:34:41 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:41 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:46 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:47 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:52 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:34:52 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:34:59 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:35:00 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:35:01 TRACE daemon::monitor: Updating status of 0 transactions
2022-02-02 03:35:05 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:35:05 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:35:10 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:35:10 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:35:19 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:35:20 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:35:21 TRACE daemon::monitor: Updating status of 0 transactions
2022-02-02 03:35:28 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:35:28 TRACE daemon::bitmex_price_feed: Received pong
2022-02-02 03:35:33 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-02-02 03:35:33 TRACE daemon::bitmex_price_feed: Received pong
```

So it can happen that we receive a `pong`, but we still don't receive a price. We should probably stop after x pongs and just establish a new connection?

**Comment by thomaseizinger** [comment]
Do you have more logs? This is quite interesting!

We are listening for binned 1m quotes so I wouldn't expect an update any more frequent than once a minute!

**Comment by bonomat** [comment]
Afair: his initial logs were only these: 

```
2022-02-02 04:16:48  INFO daemon::bitmex_price_feed: Connected to BitMex realtime API
2022-02-02 04:16:48 TRACE daemon::bitmex_price_feed: Not a 'table' message, skipping... text={"info":"Welcome to the BitMEX Realtime API.","version":"2022-01-26T07:45:06.000Z","timestamp":"2022-02-02T04:16:48.993Z","docs":"https://www.bitmex.com/app/wsAPI","limit":{"remaining":179}}
2022-02-02 04:16:48 TRACE daemon::bitmex_price_feed: Not a 'table' message, skipping... text={"success":true,"subscribe":"quoteBin1m:XBTUSD","request":{"op":"subscribe","args":"quoteBin1m:XBTUSD"}}


```

And then it started ping/pong

**Comment by thomaseizinger** [comment]
This is fixed by #1342.
